frontside/lib/views/home/home.dart

```diff
@@ -1,6 +1,69 @@
+import 'dart:async';
+
 import 'package:flutter/material.dart';
 import 'package:frontside/parts/navbar.dart';
 
+import 'package:frontside/services/places/autocomplete_response.pb.dart';
+import 'package:frontside/services/places/places.dart';
+import 'package:frontside/utils/debouncer.dart';
+
+class PlacesSearchDelegate extends SearchDelegate<AutocompleteEntry?> {
+  final _debouncer = new Debouncer();
+
+  @override
+  Widget buildLeading(BuildContext context) {
+    return IconButton(
+      icon: Icon(Icons.arrow_back),
+      onPressed: () {
+        close(context, null);
+      },
+    );
+  }
+
+  Future<AutocompleteResponse> _getResults() async {
+    final Completer<AutocompleteResponse> _completer = new Completer();
+
+    _debouncer.debounce(() async {
+      _completer.complete(await getPlaces(query));
+    });
+
+    return _completer.future;
+  }
+
+  @override
+  Widget buildResults(BuildContext context) {
+    return FutureBuilder(
+        builder: (BuildContext context,
+            AsyncSnapshot<AutocompleteResponse> response) {
+          var places = response.data?.data ?? [];
+
+          return ListView.builder(
+            itemBuilder: (context, index) {
+              var place = places[index];
+
+              return ListTile(
```

Comment:
good practice in the builders is to create widget below and there create whole widget or in case this is used in other places export it to other file, but for this case it is shot so this is fine